tests/rptest/redact_info.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python3
+
+import json
+import sys
+
+
+class Redactor:
```

Comment:
to make this work, you'd have to be scanning all the different report formats (it's not just json), _and_ the individual tests' logs (redpanda knows not to log access keys, but ducktape tests don't).

I think it would be much more robust (and perhaps less work?) to set up IAM roles https://github.com/redpanda-data/devprod/issues/409

Reply:
This change is wrt the data that we are storing in the test results DB so that it is not accessed by anyone querying the DB. Since we store `report.json` contents only, hence this change only deals with it. We are aware of devprod#409's usefulness  and will work on it as part of our unplanned epic or once we have spare cycles.

Reply:
I think I'm missing some context.  Why is it okay to have secrets in artifacts, but not in the test database?

I would have thought we would either want to avoid them everywhere, or tolerate them everywhere (as long as long as the storage location is behind some auth).

Reply:
I was looking at the other places the secrets are logged and it is in the session debug logs (`session_log.debug`) file. This is a text file:
To mask secret info here, we have two options:
- Modify https://github.com/redpanda-data/ducktape/blob/master/ducktape/command_line/main.py#L129
- Add text processor that will remove these creds (cumbersome process)

@ivotron can you please have a look at comments and suggest path forward?

Reply:
I agree with John. we should address https://github.com/redpanda-data/devprod/issues/409, and in that way https://github.com/redpanda-data/devprod/issues/452 can be closed as "not needed anymore"